pkg/client/cache/listwatch.go

```diff
@@ -34,22 +35,30 @@ type ListWatch struct {
 
 // ListWatch knows how to list and watch a set of apiserver resources.
 func (lw *ListWatch) List() (runtime.Object, error) {
-	return lw.Client.
-		Get().
-		Namespace(lw.Namespace).
-		Resource(lw.Resource).
-		SelectorParam("fields", lw.FieldSelector).
-		Do().
-		Get()
+	if lw.Client == nil {
```

Comment:
I think this is a coding error higher up the stack - no one should ever pass in a nil client.  Do you know what code path was generating this error?